server/profile.py

```diff
@@ -449,6 +449,9 @@ def get_ordered_transcript(profile_uc_dict_list):
             in s.programs and s.id not in closed_scholarship_ids_set]
     scholarships_dict = [s.to_dict() for s in scholarships]
 
+    if profile_user.id != current_user.id:
+        scholarships = []
```

Comment:
Instead of clearing the `scholarships` list after we've already done the work of populating it, can we put this `if` above the processing so that we skip it entirely if we're not on the current user's profile?